rpcs3/Emu/NP/np_requests.cpp

```diff
@@ -35,70 +35,44 @@ namespace np
 	u32 np_handler::get_server_status(SceNpMatching2ContextId ctx_id, vm::cptr<SceNpMatching2RequestOptParam> optParam, u16 server_id)
 	{
 		// TODO: actually implement interaction with server for this?
-		u32 req_id    = generate_callback_info(ctx_id, optParam);
+		u32 req_id = generate_callback_info(ctx_id, optParam, SCE_NP_MATCHING2_REQUEST_EVENT_GetServerInfo);
 		u32 event_key = get_event_key();
 
-		auto& edata                                    = allocate_req_result(event_key, SCE_NP_MATCHING2_EVENT_DATA_MAX_SIZE_GetServerInfo, sizeof(SceNpMatching2GetServerInfoResponse));
+		auto& edata = allocate_req_result(event_key, SCE_NP_MATCHING2_EVENT_DATA_MAX_SIZE_GetServerInfo, sizeof(SceNpMatching2GetServerInfoResponse));
 		SceNpMatching2GetServerInfoResponse* serv_info = reinterpret_cast<SceNpMatching2GetServerInfoResponse*>(edata.data());
-		serv_info->server.serverId                     = server_id;
-		serv_info->server.status                       = SCE_NP_MATCHING2_SERVER_STATUS_AVAILABLE;
+		serv_info->server.serverId = server_id;
+		serv_info->server.status = SCE_NP_MATCHING2_SERVER_STATUS_AVAILABLE;
 		np_memory.shrink_allocation(edata.addr(), edata.size());
 
-		const auto cb_info = take_pending_request(req_id);
-
-		if (cb_info.cb)
-		{
-			sysutil_register_cb([=, size = edata.size()](ppu_thread& cb_ppu) -> s32
-				{
-					cb_info.cb(cb_ppu, cb_info.ctx_id, req_id, SCE_NP_MATCHING2_REQUEST_EVENT_GetServerInfo, event_key, 0, size, cb_info.cb_arg);
-					return 0;
-				});
-		}
+		const auto cb_info = *take_pending_request(req_id);
```

Comment:
shouldn't this be checked for nullptr ?

Reply:
Created in the same function, it can't fail.